verification/verify.go

```diff
@@ -100,19 +105,21 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
 
 	// TODO sig checking; extract from broadcast.go
 
-	if sv.policy.Verifier == nil {
-		return nil, nil
-	}
+	var err error
+	res := &Results{}
+
 	// TODO Use policy sampling rate to determine whether to invoke verifier.
 	//      If not, exit early. Seed sample using source data for repeatability!
-	res, err := sv.policy.Verifier.Verify(params)
+	if sv.policy.Verifier != nil {
+		res, err = sv.policy.Verifier.Verify(params)
+	} else if params.Orchestrator == nil {
```

Comment:
In order to preserve the current behavior, we may need to keep it.  Either that, or something else may be wrong.

With it removed by the following patch:
```diff
diff --git a/verification/verify.go b/verification/verify.go
index 6a429060..d00aecb7 100644
--- a/verification/verify.go
+++ b/verification/verify.go
@@ -112,8 +112,6 @@ func (sv *SegmentVerifier) Verify(params *Params) (*Params, error) {
        //      If not, exit early. Seed sample using source data for repeatability!
        if sv.policy.Verifier != nil {
                res, err = sv.policy.Verifier.Verify(params)
-       } else if params.Orchestrator == nil {
-               return nil, nil
        }
 
        pxls := res.Pixels
```
a test fails.  Should we change the test or leave it as-is?  ...or is there something else we should be doing instead?  Here is the failing test:
```
$ go test
E0211 16:08:33.523832   32603 epic.go:151] Could not submit request Post : unsupported protocol scheme ""
--- FAIL: TestVerify (1.34s)
    verify_test.go:40: 
        	Error Trace:	verify_test.go:40
        	Error:      	Expected nil, but got: &verification.Params{ManifestID:"", Source:(*stream.HLSSegment)(nil), Profiles:[]ffmpeg.VideoProfile(nil), Orchestrator:(*net.OrchestratorInfo)(nil), Results:(*net.TranscodeData)(nil), URIs:[]string(nil), Renditions:[][]uint8(nil)}
        	Test:       	TestVerify
demuxer: Unable to open file
Freeing input based on OPEN INPUT error
E0211 16:08:34.866358   32603 ffmpeg.go:292] Transcoder Return : No such file or directory
demuxer: Unable to open file
Freeing input based on OPEN INPUT error
E0211 16:08:35.328943   32603 ffmpeg.go:292] Transcoder Return : No such file or directory
FAIL
exit status 1
FAIL	github.com/livepeer/go-livepeer/verification	4.118s
```

Reply:
Hm so originally the expected behavior was for a segment verifier to exit early using a policy with a nil verifier. In this PR, a segment verifier using a policy with a nil verifier is expected to count pixels and compare with the reported pixels. So, one could argue that it is reasonable for the segment verifier to return the provided `Params` since there are no rendition segments and no reported pixels so pixel count verification automatically passes. In that case, the test would just need to be adjusted to check for that expected behavior. This feels reasonable to me.

Reply:
done in 1fd94970ddf2bdbe683489e33ba8259a7ef3dc42